packages/core/src/login/oidc/IClientRegistrar.ts

```diff
@@ -58,6 +58,22 @@ function isValidUrl(url: string): boolean {
   }
 }
 
+export function determineSigningAlg(
+  supported: string[],
+  preferred: string[]
+): string | null {
+  let preferredSupportedAlg = null;
+  preferred.some((signingAlg) => {
+    if (supported.includes(signingAlg)) {
+      preferredSupportedAlg = signingAlg;
+      // returnning true breaks the .some() iteration.
+      return true;
+    }
+    return false;
+  });
+  return preferredSupportedAlg;
```

Comment:
This could be written more simply using `.find`:

```
export function determineSigningAlg(
  supported: string[],
  preferred: string[]
): string | null {
  return preferred.find((signingAlg) => {
    return supported.includes(signingAlg);
  }) || null;
}
```

Reply:
d7febeaa8dc187ada7ce826058fd42a4c1dbe7b5